### PR TITLE
feat: Implementing Counter and expose to js side

### PR DESCRIPTION
**Pull request by Leeeon233**

### CODE REVIEW

**Comment by zxch3n** [comment]
CI failed

**Comment by Leeeon233** [comment]
> CI failed

Done

**Review by zxch3n — COMMENTED** [review]
Most of the time when someone is using the counter containers, they use small integer values to inc or dec. So the current encoding schema is not very efficient, because each counter will take at least 8 bytes.

I think we can switch to using leb128 for encoding counter value when the value is int, and abs(value) < threshold. We may use a new value kind value to indicate it's f64 or a leb128.

File: crates/loro-internal/src/encoding/encode_reordered.rs [hunk 1]
```diff
@@ -1443,9 +1453,22 @@ fn decode_op(
             }
         }
         #[cfg(feature = "counter")]
-        ContainerType::Counter => {
-            crate::op::InnerContent::Future(FutureInnerContent::Counter(prop as i64))
-        }
+        ContainerType::Counter => match value {
+            Value::F64(c) => crate::op::InnerContent::Future(FutureInnerContent::Counter(c)),
+            Value::I64(c) => crate::op::InnerContent::Future(FutureInnerContent::Counter(c as f64)),
+            Value::Future(FutureValue::Unknown { kind, data }) => {
```
Comment (zxch3n) [inline crates/loro-internal/src/encoding/encode_reordered.rs:1459]: Why there is still such a case?

Comment (Leeeon233) [reply crates/loro-internal/src/encoding/encode_reordered.rs]: Oh, it's unnecessary

**Review by zxch3n — APPROVED** [review]

**Comment by zxch3n** [comment]
You can include the `changeset` in this PR directly

**Pull request merged**